### PR TITLE
[Redis] -- Add overview and sessions documentation

### PR DESCRIPTION
**Pull request by Mordil**

### CODE REVIEW

File: 4.0/docs/redis/sessions.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+```swift
+func redis<Client: RedisClient>(
+    _ client: Client,
+    storeData data: SessionData,
+    forID id: SessionID
+) -> EventLoopFuture<Void> {
+    // stores each data field as a separate hash field,
+    // with the hash key being the sesson ID
+    return client.hmset(data.storage, in: RedisKey(id.string))
```
Comment (Mordil) [inline 4.0/docs/redis/sessions.md:59]: @0xTim I know that this API isn't exposed yet, so should I just remove this section on the Hash example, and we put it in later?

File: 4.0/docs/redis/sessions.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Redis can act as a storage provider for caching [session data](../sessions.md#session-data) such as user credentials.
+
+If a custom [`RedisSessionsDelegate`](https://api.vapor.codes/redis/master/Redis/RedisSessionsDelegate/) isn't provided, a default will be used with the following behavior.
+
+## Default Behavior
+
+### SessionID Creation
+
+Unless you implement the [`makeNewID()`](https://api.vapor.codes/redis/master/Redis/RedisSessionsDelegate/#redissessionsdelegate.makeNewID()) method in [your own `RedisSessionsDelegate`](#RedisSessionsDelegate), all [`SessionID`](https://api.vapor.codes/vapor/master/Vapor/SessionID/) values will be created by doing the following:
```
Comment (0xTim) [inline 4.0/docs/redis/sessions.md:11]: Is this true if using the default sessions middleware?

**Review by 0xTim — CHANGES_REQUESTED** [review]
A few comments and queries

File: 4.0/docs/sessions.md [hunk 3]
```diff
@@ -71,7 +71,23 @@ Finally, add `SessionRecord`'s migration to your database's migrations. This wil
...
 ```
 
-Make sure to run your application's migrations after adding the new migration. Sessions will now be stored in your application's database allowing them to persist between restarts and be shared between multiple instances of your app. 
+Make sure to run your application's migrations after adding the new migration. Sessions will now be stored in your application's database allowing them to persist between restarts and be shared between multiple instances of your app.
+
+### Redis
+
+Redis provides support for storing session data in your configured Redis instance. This section assumes you have [configured Redis](redis/overview.md) and can send commands to the Redis instance.
+
+The first step for enabling the Redis sessions driver:
```
[the remark this answers is not in the document]
Comment (Mordil) [reply 4.0/docs/sessions.md:80]: 🤦 This isn't a nit, this is just correcting poor guide writing haha

File: 4.0/docs/redis/sessions.md (hunk 2 above)
Comment (Mordil) [reply 4.0/docs/redis/sessions.md:11]: Yes, there is a [default implementation](https://github.com/vapor/redis/blob/master/Sources/Redis/Redis%2BSessions.swift#L27) of the `makeNewID()` protocol requirement on the `RedisSessionsDelegate` itself, and the [`DefaultSessionsDriverDelegate`](https://github.com/vapor/redis/blob/master/Sources/Redis/Redis%2BSessions.swift#L92) only implements the required methods.

Why? Is this horrible semantics?

Comment (0xTim) [reply 4.0/docs/redis/sessions.md:11]: Oh hang on, is this just the name of the session in Redis? I was getting confused between the name of the cookie. In which case this is fine

**Review by 0xTim — APPROVED** [review]
@Mordil this looks good to go. I'll leave it open, as soon as vapor/redis is ready to be tagged feel free to merge the PR and then link to the docs in the release notes

File: 4.0/docs/redis/sessions.md (hunk 2 above)
Comment (Mordil) [reply 4.0/docs/redis/sessions.md:11]: No it is the name of the session itself, not just the Redis key.

`makeNewID` is called when the `SessionDriver.createSession` method is called

https://github.com/vapor/redis/blob/5255824185c836aef5514d1d9e571571653d8569/Sources/Redis/Redis%2BSessions.swift#L70

Now that you’re bringing it up, it sounds like this should be the token itself but only the Redis key.

I’ll patch Redis and update these docs

Comment (Mordil) [reply 4.0/docs/redis/sessions.md:11]: Redis patch(es):
- https://github.com/vapor/redis/pull/176
- https://github.com/vapor/redis/pull/177

**Comment by Mordil** [comment]
@0xTim It looks like I don't have write access to the `master` branch.

> You’re not [authorized](https://docs.github.com/articles/about-protected-branches/) to merge this pull request.

**Comment by 0xTim** [comment]
@Mordil permissions should be correct now

**Review by 0xTim — APPROVED** [review]

**Pull request merged**